### ISSUE TITLE
Extra Utilities 2 Unexpected World Change

### ISSUE THREAD

**Issue by limbo-app** [SpongePowered/SpongeForge#1997]
This issue was originally submitted by @rojo8399 in [SpongePowered/SpongeAPI#1776](https://github.com/SpongePowered/SpongeAPI/issues/1776).

Edited missing information and replies into the OP - @ryantheleach
___

SpongeForge: 1.12.2-2586-7.1.0-BETA-2897
forge: 1.12.2-14.23.2.2611
java: 8
OS: Win 10 latest

When breaking any compressed blocks from extra utilities 2, this is spammed in console. It doesn't happen on the break. It happens while it's breaking, so even just tapping it causes this.
Don't know if it's an EU2 issue or sponge.

Here's the whole error.

```
[Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /*                                                                                                                              Unexpected World Change Detected!                                                                                                                               */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /* Sponge's tracking system is very dependent on knowing when
a change to any world takes place, however there are chances
where Sponge does not know of changes that mods may perform.
In cases like this, it is best to report to Sponge to get this
change tracked correctly and accurately. */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /* StackTrace:                                                                                                                                                                                                                                                                                  */
[08:22:28] [Server thread/ERROR] [Sponge]: /* java.lang.Exception: null                                                                                                                                                                                                                                                                    */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.printUnexpectedBlockChange(PhaseTracker.java:450)                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:582)                                                                                                                                                                                                */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:552)                                                                                                                                                                                                */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.world.WorldServer.func_180501_a(WorldServer.java:2710)                                                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     com.rwtema.extrautils2.blocks.BlockCompressed.func_176195_g(BlockCompressed.java:108)                                                                                                                                                                                                    */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185887_b(BlockStateContainer.java:403)                                                                                                                                                                            */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     jackyy.avaritiatweaks.tweaks.ModEventsHandler.onLeftClick(ModEventsHandler.java:60)                                                                                                                                                                                                      */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler_754_ModEventsHandler_onLeftClick_LeftClickBlock.invoke(.dynamic)                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)                                                                                                                                                                                               */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:651)                                                                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeForgeEventFactory.createPlayerInteractEvent(SpongeForgeEventFactory.java:877)                                                                                                                                                                          */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(SpongeForgeEventFactory.java:524)                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:310)                                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:367)                                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:422)                                                                                                                                                                                                      */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:207)                                                                                                                                                                                                                       */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.SpongeCommonEventFactory.callInteractBlockEventPrimary(SpongeCommonEventFactory.java:613)                                                                                                                                                                 */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.handler$onServerTickEnd$zjf000(MinecraftServer.java:3860)                                                                                                                                                                                           */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:711)                                                                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)                                                                                                                                                                                                                       */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     java.lang.Thread.run(Unknown Source)                                                                                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
```

Remapped: https://gist.github.com/ryantheleach/5fb0a2d060d2b0e87f7ae1422707b4a8

**Comment by rojo8399** [comment]
Here's the whole error.

`[Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /*                                                                                                                              Unexpected World Change Detected!                                                                                                                               */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /* Sponge's tracking system is very dependent on knowing when
a change to any world takes place, however there are chances
where Sponge does not know of changes that mods may perform.
In cases like this, it is best to report to Sponge to get this
change tracked correctly and accurately. */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/
[08:22:28] [Server thread/ERROR] [Sponge]: /* StackTrace:                                                                                                                                                                                                                                                                                  */
[08:22:28] [Server thread/ERROR] [Sponge]: /* java.lang.Exception: null                                                                                                                                                                                                                                                                    */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.printUnexpectedBlockChange(PhaseTracker.java:450)                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:582)                                                                                                                                                                                                */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:552)                                                                                                                                                                                                */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.world.WorldServer.func_180501_a(WorldServer.java:2710)                                                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     com.rwtema.extrautils2.blocks.BlockCompressed.func_176195_g(BlockCompressed.java:108)                                                                                                                                                                                                    */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185887_b(BlockStateContainer.java:403)                                                                                                                                                                            */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     jackyy.avaritiatweaks.tweaks.ModEventsHandler.onLeftClick(ModEventsHandler.java:60)                                                                                                                                                                                                      */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler_754_ModEventsHandler_onLeftClick_LeftClickBlock.invoke(.dynamic)                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)                                                                                                                                                                                               */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:651)                                                                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeForgeEventFactory.createPlayerInteractEvent(SpongeForgeEventFactory.java:877)                                                                                                                                                                          */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(SpongeForgeEventFactory.java:524)                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:310)                                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:367)                                                                                                                                                                                                   */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:422)                                                                                                                                                                                                      */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:207)                                                                                                                                                                                                                       */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event.SpongeCommonEventFactory.callInteractBlockEventPrimary(SpongeCommonEventFactory.java:613)                                                                                                                                                                 */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.handler$onServerTickEnd$zjf000(MinecraftServer.java:3860)                                                                                                                                                                                           */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:711)                                                                                                                                                                                                              */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)                                                                                                                                                                                                                       */
[08:22:28] [Server thread/ERROR] [Sponge]: /*     java.lang.Thread.run(Unknown Source)                                                                                                                                                                                                                                                     */
[08:22:28] [Server thread/ERROR] [Sponge]: /************************************************************************************************************************************************************************************************************************************************************************************************/`

**Comment by Keaton1188** [comment]
- SpongeForge version: ?
- Forge version: ?
- Java version: ?
- Operating System: ?

**Comment by rojo8399** [comment]
SpongeForge: 1.12.2-2586-7.1.0-BETA-2897
forge: 1.12.2-14.23.2.2611
java: 8
OS: Win 10 latest

**Comment by ryantheleach** [comment]
```
   public float getBlockHardness(IBlockState state, World worldIn, BlockPos pos) {
      int value = ((Integer)worldIn.getBlockState(pos).getValue(this.property_compression)).intValue();
      IBlockState oldState = worldIn.getBlockState(pos);

      try {
         worldIn.setBlockState(pos, this.baseState, 4);
         float err = this.baseState.getBlockHardness(worldIn, pos) * (float)Math.pow(2.25D, (double)(value - 1));
         worldIn.setBlockState(pos, oldState, 4);
         return err;
      } catch (Throwable var7) {
         worldIn.setBlockState(pos, oldState, 4);
         throw Throwables.propagate(var7);
      }
   }
```

**Comment by phit** [comment]
@rojo8399 is this still an issue on latest?

**Comment by ronaldburns** [comment]
Still happening on 1.12.2-2611-7.1.0-BETA-2929
Do you mean API 8?

```
/************************************************************************************************************************************************************************************************************************************************************************************************/
/* Sponge's tracking system is very dependent on knowing when
a change to any world takes place, however there are chances
where Sponge does not know of changes that mods may perform.
In cases like this, it is best to report to Sponge to get this
change tracked correctly and accurately. */
/************************************************************************************************************************************************************************************************************************************************************************************************/
/* StackTrace:                                                                                                                                                                                                                                                                                  */
/* java.lang.Exception: null                                                                                                                                                                                                                                                                    */
/*     org.spongepowered.common.event.tracking.PhaseTracker.printUnexpectedBlockChange(PhaseTracker.java:441)                                                                                                                                                                                   */
/*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:573)                                                                                                                                                                                                */
/*     org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:543)                                                                                                                                                                                                */
/*     net.minecraft.world.WorldServer.func_180501_a(WorldServer.java:2710)                                                                                                                                                                                                                     */
/*     com.rwtema.extrautils2.blocks.BlockCompressed.func_176195_g(BlockCompressed.java:110)                                                                                                                                                                                                    */
/*     net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185887_b(BlockStateContainer.java:403)                                                                                                                                                                            */
/*     jackyy.avaritiatweaks.tweaks.ModEventsHandler.onLeftClick(ModEventsHandler.java:60)                                                                                                                                                                                                      */
/*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler_753_ModEventsHandler_onLeftClick_LeftClickBlock.invoke(.dynamic)                                                                                                                                                              */
/*     net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)                                                                                                                                                                                               */
/*     net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:651)                                                                                                                                                                                                              */
/*     org.spongepowered.mod.event.SpongeForgeEventFactory.createPlayerInteractEvent(SpongeForgeEventFactory.java:879)                                                                                                                                                                          */
/*     org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(SpongeForgeEventFactory.java:526)                                                                                                                                                                                     */
/*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:307)                                                                                                                                                                                                   */
/*     org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:364)                                                                                                                                                                                                   */
/*     org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:422)                                                                                                                                                                                                      */
/*     org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:207)                                                                                                                                                                                                                       */
/*     org.spongepowered.common.event.SpongeCommonEventFactory.callInteractBlockEventPrimary(SpongeCommonEventFactory.java:702)                                                                                                                                                                 */
/*     net.minecraft.server.MinecraftServer.handler$onServerTickEnd$zjf000(MinecraftServer.java:3860)                                                                                                                                                                                           */
/*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:711)                                                                                                                                                                                                              */
/*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)                                                                                                                                                                                                                       */
/*     java.lang.Thread.run(Unknown Source)                                                                                                                                                                                                                                                     */
/************************************************************************************************************************************************************************************************************************************************************************************************/
```

**Comment by phit** [comment]
no, 7.1.0 is the only stable version right now
thanks for testing again, I was just hoping this was fixed with https://github.com/SpongePowered/SpongeForge/issues/2010

**Comment by gabizou** [comment]
Looks like I really do need to enter phase states for plugin listeners, much like how stack frames are creating for event listeners. I can write something for stable-7 which should work out fine.

**Comment by phit** [comment]
appears to be resolved running eed4b540aeff1dd1be8670da969f9c525ddd31dc and `extrautils2-1.12-1.7.5`
